### PR TITLE
[BugFix] [310p] Fix attention accuracy issue

### PR DESCRIPTION
**Pull request by pu-zhe**
### What this PR does / why we need it?
This pull request resolves an attention accuracy issue by enhancing the AttentionMaskBuilder310 to correctly handle the maximum model length. The change ensures that the attention mask generation process is properly parameterized by the model's configuration, rather than relying on a fixed internal value. This leads to more accurate attention mask creation, which is crucial for the correct functioning of the attention mechanism.
Update fused_moe to main branch.
### Does this PR introduce _any_ user-facing change?
No
### How was this patch tested?
Qwen3 dense mode & moe model e2e test
- vLLM version: v0.15.0
- vLLM main: https://github.com/vllm-project/vllm/commit/83b47f67b1dfad505606070ae4d9f83e50ad4ebd

### CODE REVIEW

**Comment by pu-zhe** [comment]
@gemini-code-assist summary

File: tests/ut/_310p/attention/test_attention_mask_310.py [hunk 1]
```diff
@@ -47,7 +48,7 @@ def test_get_swa_mask_310(self, mock_format_cast):
 
         sliding_window = 128
         swa_mask = self.attention_mask_builder.get_swa_mask(torch.float16, sliding_window)
-        self.assertEqual(swa_mask.shape, (1, 128, 2048, 16))
+        self.assertEqual(swa_mask.shape, (1, self.max_seqlen // 16, self.max_seqlen, 16))
```
Comment (ZhaoJiangJiang) [inline tests/ut/_310p/attention/test_attention_mask_310.py]: What does the "16" represent here? Is it intended to represent the BLOCK_SIZE？
